src/api-service/__app__/onefuzzlib/endpoint_authorization.py

```diff
@@ -43,6 +44,38 @@ def is_agent(token_data: UserInfo) -> bool:
     return False
 
 
+def can_modify_config(req: func.HttpRequest, config: InstanceConfig) -> bool:
+    user_info = parse_jwt_token(req)
+    if not isinstance(user_info, UserInfo):
+        False
+
+    # make mypy happy
+    assert isinstance(user_info, UserInfo)
+
+    if config.admins is None:
+        return True
+
+    return user_info.object_id in config.admins
+
+
+def can_modify_pools(req: func.HttpRequest) -> bool:
```

Comment:
can we be specific here and distinguish between "no admins specified" and "The user is not an authorized admin". This will allows us give a more meaningful message when the access fail.
maybe if we return a `Union[bool, AuthtenticationError]`

Reply:
addressed.